Tests/Integration/Network/Connection/FogSyncErrorConnectionIntTests.swift

```diff
@@ -0,0 +1,266 @@
+        }
+    }
+    
+    func fogAheadConsensus(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 321
+        let consensusIndex = fogIndex - 2
+        
+        // Fog ahead of Consensus should succeed (occurs when cached Consensus block info is used)
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogBehindConsensusWithinThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusWithinThresholdAheadConsensus(
+                    transportProtocol: $0,
+                    expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusWithinThresholdAheadConsensus(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 5234523462456
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold - 2
+        
+        // Fog behind but within threshold should succeed
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogBehindConsensusAtThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusAtThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusAtThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 60
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold - 1
+        
+        // Fog behind at threshold, should fail
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: false
+        )
+    }
+    
+    func testFogBehindConsensusOverThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusOverThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusOverThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 410
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold
```

Comment:
I would have expected this to mean "AtThreshold" instead of "OverThreshold" where you're adding the threshold value,  so when the sync checker is subtracting and getting the delta, it will be == threshold (i.e. at threshold), no?

Reply:
updated